torch/testing/_internal/common_device_type.py

```diff
@@ -942,13 +944,19 @@ def _parametrize_test(self, test, generic_cls, device_cls):
                     dtypes = {next(dtype for dtype in ANY_DTYPE_ORDER if dtype in supported)}
                 else:
                     dtypes = {}
-
+            elif self.opinfo_dtypes == OpDTypes.any_common_cpu_xpu_one:
+                # Tries to pick a dtype that supports both CPU and CUDA
+                supported = set(op.dtypes).intersection(op.dtypesIfXPU)
+                if supported:
+                    dtypes = {next(dtype for dtype in ANY_DTYPE_ORDER if dtype in supported)}
+                else:
+                    dtypes = {}
             elif self.opinfo_dtypes == OpDTypes.none:
                 dtypes = {None}
             else:
                 raise RuntimeError(f"Unknown OpDType: {self.opinfo_dtypes}")
 
-            if self.allowed_dtypes is not None:
+            if self.allowed_dtypes is not None and dtypes is not None:
```

Comment:
Why do we need to check `dtypes` is not None?

Reply:
Usually it could not help, but if the code goes to 'else' path, https://github.com/daisyden/pytorch/blob/daisyden/stock_pt/torch/testing/_internal/common_device_type.py#L953 

```
elif self.opinfo_dtypes == OpDTypes.any_common_cpu_xpu_one:
                # Tries to pick a dtype that supports both CPU and CUDA
                supported = set(op.dtypes).intersection(op.dtypesIfXPU)
                if supported:
                    dtypes = {next(dtype for dtype in ANY_DTYPE_ORDER if dtype in supported)}
                else:
                    dtypes = {}
```
An error could occur.
```
>>> dtypes={}
>>> dtypes.intersection({torch.float16,})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'dict' object has no attribute 'intersection'
```

Reply:
as we have default dtypesIfXPU now, we should not go to the else path.